chemprop/args.py

```diff
@@ -566,174 +673,282 @@ def process_args(self) -> None:
+            if self.dataset_type == "classification":
+                self.metric = "auc"
+            elif self.dataset_type == "multiclass":
+                self.metric = "cross_entropy"
+            elif self.dataset_type == "spectra":
+                self.metric = "sid"
+            elif self.dataset_type == "regression" and self.loss_function == "bounded_mse":
+                self.metric = "bounded_mse"
+            elif self.dataset_type == "regression":
+                self.metric = "rmse"
             else:
-                raise ValueError(f'Dataset type {self.dataset_type} is not supported.')
+                raise ValueError(f"Dataset type {self.dataset_type} is not supported.")
 
         if self.metric in self.extra_metrics:
-            raise ValueError(f'Metric {self.metric} is both the metric and is in extra_metrics. '
-                             f'Please only include it once.')
+            raise ValueError(
+                f"Metric {self.metric} is both the metric and is in extra_metrics. "
+                f"Please only include it once."
+            )
 
         for metric in self.metrics:
-            if not any([(self.dataset_type == 'classification' and metric in ['auc', 'prc-auc', 'accuracy', 'binary_cross_entropy', 'f1', 'mcc']), 
-                    (self.dataset_type == 'regression' and metric in ['rmse', 'mae', 'mse', 'r2', 'bounded_rmse', 'bounded_mae', 'bounded_mse']), 
-                    (self.dataset_type == 'multiclass' and metric in ['cross_entropy', 'accuracy', 'f1', 'mcc']),
-                    (self.dataset_type == 'spectra' and metric in ['sid','wasserstein'])]):
-                raise ValueError(f'Metric "{metric}" invalid for dataset type "{self.dataset_type}".')
-        
+            if not any(
+                [
+                    (
+                        self.dataset_type == "classification"
+                        and metric
+                        in ["auc", "prc-auc", "accuracy", "binary_cross_entropy", "f1", "mcc"]
+                    ),
+                    (
+                        self.dataset_type == "regression"
+                        and metric
+                        in [
+                            "rmse",
+                            "mae",
+                            "mse",
+                            "r2",
+                            "bounded_rmse",
+                            "bounded_mae",
+                            "bounded_mse",
+                        ]
+                    ),
+                    (
+                        self.dataset_type == "multiclass"
+                        and metric in ["cross_entropy", "accuracy", "f1", "mcc"]
+                    ),
+                    (self.dataset_type == "spectra" and metric in ["sid", "wasserstein"]),
+                ]
+            ):
+                raise ValueError(
+                    f'Metric "{metric}" invalid for dataset type "{self.dataset_type}".'
+                )
+
```

Comment:
I don't have a specific formatting alternative, but this is illegible. Please revert to what was there before (I know the flake8 formatter is just doing the best it can).